### PR TITLE
Feature/doc-page-responsiveness

### PR DESCRIPTION
**Pull request by yuki-tylar**

### CODE REVIEW

**Review by pmvald — CHANGES_REQUESTED** [review]
Looks good. I made a few suggestion regarding naming conventions.

There are several tricky situations: for example:

you are in mobile size -> open the sidebar -> extend the window to desktop size -> reduce to mobile size again -> the sidebar is still there 

it is expected that upon reducing size to mobile version left sidebar disappears. But this cannot be done by css alone. We need to set js to listen to widnow size change and set the state variable "showLeftSidebar" to false when we cross the threshold ...So please open an issue tagged "bug" for this scenario (include a brief description), we will fix it later ...

**Comment by pmvald** [comment]
Just ignore the comment about header and overlay ... our design is different

**Comment by yuki-tylar** [comment]
modified as below

### doc-page-res.tsx
* change state `showLeftSidebar` to `false` when user changes window size to bigger than 768px

### project.scss
* merge all media query in one place 
* use more variables for better customization (`$leftSideBarWidth` / `$animationPeriod` / ...)
* change class name following BEM
* marge animation state in one definition (`.show-left-sidebar`)

File: src/pages/__mocks/doc-page-res.tsx [hunk 1]
```diff
@@ -45,6 +45,23 @@ const style: { [k: string]: CSSProperties } = {
...
+        timer = setTimeout(() => {
+            if(window.innerWidth >= threasholdToChangeLayout && showLeftSidebar) {
+                setShowLeftSidebar(false);
+            }    
+        }, 300);
+    }
+
+    useEffect(() => {
+        window.addEventListener('resize', windowResizeHandler, true);
+        return window.removeEventListener('resize', windowResizeHandler, true);
```
Comment (pmvald) [inline src/pages/__mocks/doc-page-res.tsx]: FYI, use effect should return a function, so it should be `return ()=>windows.....`

File: src/pages/__mocks/doc-page-res.tsx [hunk 2]
```diff
@@ -45,6 +45,23 @@ const style: { [k: string]: CSSProperties } = {
 export default function() {
 
     const [showLeftSidebar, setShowLeftSidebar] = useState(false);
+    
+    let timer: any;
+    const threasholdToChangeLayout = 768;
+
+    const windowResizeHandler = () => {
```
Comment (pmvald) [inline src/pages/__mocks/doc-page-res.tsx]: callback should move inside useEffect, other wise you will have "stale state problem"; summary: React component function is called every time you change the state, that means the function you define inside the function body will become "invalid" upon state change (like when you press the toggle button) - So the even will call an invalid callback and nothing will happen. I moved that inside the useEffect and now it stays on even after state change .. React becomes very tricky in these situations, either way, it works now

**Review by pmvald — APPROVED** [review]
Please read the comments and resolve them, it is merged now

**Comment by pmvald** [comment]
Sounds good, you also need to resolve the conversations (press resolve button for each review), and then it is possible to merge - this time I do it myself

**Comment by pmvald** [comment]
This time I resolve the discussions, but please go through the new comments and take action (one commens needs you to create an issue for future improvement ...)

**Pull request merged**